### DISCUSSION TITLE
Why are there many different OS-specific releases? Actual difference between them?

### DISCUSSION THREAD

**Discussion by MikeDombo** [emqx/emqx#6916] [Q&A] [answered]
There are many many different releases with precompiled binaries by OS and OS version. I had a brief look at two examples and I found that the emqx binary itself is exactly the same, but the included dynamic libraries were different. Is that the only difference? 
If so, is it possible to configure emqx to compile with static libraries and/or using system libraries in order to have a single release that will work for all (or at least many) Linux OSs?

**Comment by zmstone** [comment, marked as answer]
Hi @MikeDombo 

Yes. The c/c++ artifacts are the reason why we build EMQ X on different OS.
Erlang runtime (the vm and tools), OpenSSL, glibc, and quite a few others.

I’m not sure if anyone from Erlang community actually managed to link everything statically.

**Reply by MikeDombo** [reply]
Would `--enable-static-nifs` work to resolve this? From my reading on the matter it seems like it might, but not very clear.

**Reply by zmstone** [reply]
I am not very clear either, `--enable-static-nifs` is one piece of the puzzle.
I will have to experiment it to have a better understanding of all the dependencies.
